colossalai/testing/comparison.py

```diff
@@ -13,7 +13,7 @@ def assert_not_equal(a: Tensor, b: Tensor):
 
 
 def assert_close(a: Tensor, b: Tensor, rtol: float = 1e-5, atol: float = 1e-8):
-    assert torch.allclose(a, b, rtol=rtol, atol=atol), f'expected a and b to be close but they are not, {a} vs {b}'
+    assert torch.allclose(a.to(torch.float), b.to(torch.float), rtol=rtol, atol=atol), f'expected a and b to be close but they are not, {a} vs {b}'
```

Comment:
you need to add two flags for this function. one is `same_type`, if it is true, you turn them into the same type. other is `same_device`, if it is true, you turn them into the same device.